meson.build

```diff
@@ -0,0 +1,26 @@
+project('gpgi', 'c', 'cython', meson_version: '>=1.5.0')
+
+py = import('python').find_installation(pure: false)
+
+min_numpy_version = '1.25.0'  # keep in sync with pyproject.toml
+cy = meson.get_compiler('cython')
+cython = find_program(cy.cmd_array()[0])
+
+
+# https://github.com/mesonbuild/meson/issues/9598
+incdir_numpy = run_command(py,
```

Comment:
For the fancier version, you can copy the code block at https://github.com/PyWavelets/pywt/blob/becef5486174c77727ea2a60e6392744b2cc1d4a/pywt/_extensions/meson.build#L13

Once we get to numpy >=2.0 only (one more year ....), it becomes simply:
```meson
numpy_dep = dependency('numpy')
```

Reply:
One other advantage is that the longer code block takes care of in-tree venv's . Meson by default doesn't like to see `include_directories('abspath-within-source-tree')`, which is usually a mistake, but in-tree venv's overlap with this.

Reply:
There's also a build warning related to `NPY_NO_DEPRECATED_API` that you can take care of with the code linked above. Warning-free builds are one of the upsides of moving to meson-python:)

The `define_macros` thing in `setuptools` has more generic alternatives in Meson. In this case, a `c_args` keyword for passing any argument to a C compiler. In the case of `NPY_TARGET_VERSION` (which doesn't need setting I think, but it can't hurt either): `c_args: ['-DNPY_TARGET_VERSION=NPY_1_25_API_VERSION']`

Reply:
> Once we get to numpy >=2.0 only (one more year ....), it becomes simply:

just to be sure, you mean that the `required: false` argument can be removed once numpy 1 is dropped ?

Reply:
No I mean the 25 lines of code at https://github.com/PyWavelets/pywt/blob/becef5486174c77727ea2a60e6392744b2cc1d4a/pywt/_extensions/meson.build#L13-L43 can be replaced by the one-liner then.

To support that, I had to add the `numpy-config` executable, and that only landed in NumPy 2.0

Reply:
When dropping the `required: false` keyword argument, the remainder of those 25 lines can be removed due to dead code elimination. Numpy 2 will always be detectable via `dependency('numpy', required: true)` (the default is true) and this `.found()` can never evaluate to false so the code block becomes unreachable.